frontend/routes/viewerCanvas/viewerCanvas.component.tsx

```diff
@@ -121,13 +131,34 @@ export class ViewerCanvas extends React.PureComponent<IProps, any> {
 		if (prev.length === 0 && curr.length > 0) {
 			viewer.mapStart();
 		}
+	}
+
+	public async renderMeasurements(prev: any[], curr: any[], highlighted: any[] = []) {
+		const { viewer } = this.props;
+
+		const toAdd = differenceBy(curr, prev, 'uuid', 'color');
+		const toRemove = differenceBy(prev, curr, 'uuid', 'color');
 
+		await viewer.removeMeasurements(toRemove);
+		await viewer.addMeasurements(toAdd, true);
+		await viewer.selectMeasurements(highlighted);
```

Comment:
why do we need to process highlights here when we also have `renderMeasurementsHighlights` that is called immediately after?

Reply:
I did it because of race conditions; it might be that the highlighted shapes are being set before the measurement has been added to the measurement. In that case the shape wouldnt show its selection. Maybe there is a way to get rid of this condition in the selector, only selecting the shapes that have already changed in the previous selector.